include/extractor/guidance/turn_analysis.hpp

```diff
@@ -82,10 +106,9 @@ class TurnAnalysis
     //
     // The treatment results in a straight turn angle of 180º rather than a turn angle of approx
     // 160
-    std::vector<TurnCandidate>
-    mergeSegregatedRoads(const NodeID from_node,
-                         const EdgeID via_eid,
-                         std::vector<TurnCandidate> turn_candidates) const;
+    std::vector<ConnectedRoad> mergeSegregatedRoads(const NodeID from_node,
+                                                    const EdgeID via_eid,
+                                                    std::vector<ConnectedRoad> intersection) const;
```

Comment:
I tried to follow your guidance for inplace modification + move in this pull request. So yes, it should be this way.